LLMNeedleHaystackTester.py

```diff
@@ -1,53 +1,54 @@
 from dotenv import load_dotenv
 import os
-import tiktoken
 import glob
 import json
 from langchain.evaluation import load_evaluator
 from langchain.chat_models import ChatOpenAI
-from anthropic import AsyncAnthropic, Anthropic
-from dotenv import load_dotenv
+
 import numpy as np
-from openai import AsyncOpenAI
+
 import asyncio
 from asyncio import Semaphore
 from datetime import datetime, timezone
 import time
 
+from abc import ABC, abstractmethod
+
 load_dotenv()
 
-class LLMNeedleHaystackTester:
+
+class LLMNeedleHaystackTester(ABC):
     """
     This class is used to test the LLM Needle Haystack.
     """
+
     def __init__(self,
                  needle="\nThe best thing to do in San Francisco is eat a sandwich and sit in Dolores Park on a sunny day.\n",
                  haystack_dir="PaulGrahamEssays",
                  retrieval_question="What is the best thing to do in San Francisco?",
-                 results_version = 1,
-                 context_lengths_min = 1000,
-                 context_lengths_max = 200000,
-                 context_lengths_num_intervals = 35,
-                 context_lengths = None,
-                 document_depth_percent_min = 0,
-                 document_depth_percent_max = 100,
-                 document_depth_percent_intervals = 35,
-                 document_depth_percents = None,
-                 document_depth_percent_interval_type = "linear",
-                 model_provider = "OpenAI",
-                 openai_api_key=None,
-                 anthropic_api_key = None,
-                 model_name='gpt-4-1106-preview',
-                 num_concurrent_requests = 1,
-                 save_results = True,
-                 save_contexts = True,
-                 final_context_length_buffer = 200,
-                 seconds_to_sleep_between_completions = None,
-                 print_ongoing_status = True):
-        """        
+                 substr_validation_words=['dolores', 'sandwich'],
```

Comment:
Do we need all those arguments? Are they ever used by someone? 🙂

Consider moving them to a constants.